wallet-new/test/DevelopmentSpec.hs

```diff
@@ -0,0 +1,78 @@
+{-# OPTIONS_GHC -fno-warn-orphans       #-}
+
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE NoImplicitPrelude          #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+-- Spec for testing `development` endpoints
+module DevelopmentSpec where
+
+import           Universum
+
+import           Data.Default (def)
+import           Pos.Client.KeyStorage (addSecretKey, getSecretKeysPlain)
+
+import           Pos.Util.BackupPhrase (BackupPhrase (..), safeKeysFromPhrase)
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Launcher (HasConfigurations)
+import           Pos.Util.QuickCheck.Property (assertProperty)
+
+import           Test.Pos.Wallet.Web.Mode (walletPropertySpec)
+import           Test.Hspec (Spec, describe)
+import           Test.Hspec.QuickCheck (modifyMaxSuccess)
+import           Test.Pos.Configuration (withDefConfigurations)
+
+import           Cardano.Wallet.API.Development.LegacyHandlers (deleteSecretKeys)
+import           Cardano.Wallet.Server.CLI (RunMode (..))
+import           Servant
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+spec :: Spec
+spec =
+    withCompileInfo def $
+    withDefConfigurations $
+    describe "development endpoint" $
+        describe "secret-keys" $ modifyMaxSuccess (const 10) deleteAllSecretKeysSpec
+
+deleteAllSecretKeysSpec :: (HasCompileInfo, HasConfigurations) => Spec
```

Comment:
These tests are fine for now but notice how they are not exerting the API externally, but relying on internal functions from the old wallet, which we are in the process of removing.

Reply:
> relying on internal functions from the old wallet
Sure, because this ^ and most of others `V1` endpoints are currently based on V0 implementation. They are mostly just wrapper around `V0` functions. I don't see a way to test those in other way as long we do have these `V0` dependencies.